examples/predict_on_grid_coordinates.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2017 The Verde Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This code is part of the Fatiando a Terra project (https://www.fatiando.org)
+#
+"""
+Gridding on a preexisting grid
+==============================
+
+The ``grid`` method of any Verde gridder allows us to interpolate the data on
+a regular grid. By passing the ``shape`` or ``spacing`` arguments, the method
+will build the regular grid by itself.
+Nevertheless, if we want to interpolate the data on a predefined regular grid,
+we can pass the grid ``coordinates`` instead.
+"""
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import pyproj
+import numpy as np
+import verde as vd
+
+# Define a CheckerBoard gridder
+region = (0, 5000, -5000, 0)
+gridder = vd.datasets.CheckerBoard(region=region)
+
+# Define a set of grid coordinates as 1d arrays
+easting = np.linspace(*region[:2], 50)
+northing = np.linspace(*region[2:], 50)
+coordinates = (easting, northing)
+
+# Use the .grid() method to predict on the grid coordinates
+grid = gridder.grid(coordinates)
```

Comment:
Using `coordinates` for the grid coordinates is going to be confusing. Probably best to use `coordinates` for data coordinates and `grid_coordinates` for grid coordinates throughout the code and examples.

Reply:
I agreed at first, but then I found that we already use `grid_coordinates` in Verde for the function that creates the coords of a regular grid. Should we use a different name for the coordinates argument?

Reply:
Yeah, you're right. `grid_coords` is reasonable I guess

Reply:
I'm having second thoughts on this. Is really an issue of having `coordinates` argument on methods like `predict` and in `grid` while the requirements are slightly different? I think having a common `coordinates` arguments for methods and functions in Verde is a feature and not an issue:
- I can autocomplete on every call by just typing `coor<TAB>` (with `grid_coords` it wouldn't work, I would need to delete and start typing `gri`).
- On some places it doesn't mind if the passed coordinates are scattered or regular gridded (forward modelling in Harmonica for instance). So those cases would be an exception where `coordinates` should be used, although grid coordinates could also be passed.
- On the specific `BaseGridder` we would have two methods that do almost the same but for different kind of coordinates: `predict` for scattered coordiantes, `grid` for grid coordinates. I think that difference is sufficient for differentiating the distribution of the coords.

One last argument, if we use `grid_coords`, an simple example of the `grid` method would look like:
```
grid = gridder.grid(grid_coords=...)
```
I can see myself struggling to read that line out loud when teaching.

What do you think?

Reply:
You're absolutely right. Typing `spline.grid(grid_coords...)` is awkward. So here is a proposal:

1. Function arguments are `coordinates` unless you need to specify 2 sets (like `distance_mask`).
2. For examples: `coordinates` means data coordinates and `grid_coords` means grid coordinates since there will usually be the 2 sets and we'll need to distinguish between them. That way the call could be `spline.grid(coordinates=grid_coords...` or `spline.grid(coordinates=vd.grid_coordinates...)`.

Reply:
Now that I write this, I'm not entire happy with requiring giving coordinates all the time. But I don't know what else to do. I would want to write some examples using this syntax to see how it feels but my feeling is that it will be a bit of a blow to general usability.

Reply:
I agree with the two points above. Just a reminders for the future: _don't underestimate the old design decisions, there might be a reason why we took that road even if we don't remember it_.

Feel free to try it out on examples, there's no rush for this PR. If we are going to change the default behavior of the base class for Verde gridders, it's good to put a lot of thinking into it.

On the note about passing coordinates every time I think we are already doing it, but hiding it through the `spacing`, `shape`, `region`, `projection` arguments. So the `grid` method currently does two things: creates a set of grid coordinates and predicts on them. I would like to split it in two:
1. Leave the coordinate generation to the user. This way they can corroborate that the grid is the right one before they have to wait until the prediction is carried out. Or they can just use a predefined grid and skip this step.
2. Use the `grid` method to predict on the given coordinates. It should also project the coords before running the prediction if any `projection` is passed. This simplifies the usage of the `grid` method, makes it easier to generalize to gridders like EQL with `easting`, `northing` and `height` (they can be part of the `coordinates` directly) and makes it easier to understand what the `projection` callback will be used for (I always struggle to think if the projection is applied _before_ the prediction or _after_ it and which coordinates should I use for `region` and `spacing`).

Finally, I think this new `grid` method will be _doing a single task, and doing it well_ 😁 .